sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OtlpConfigUtil.java

```diff
@@ -32,11 +32,9 @@ final class OtlpConfigUtil {
   static final String PROTOCOL_HTTP_PROTOBUF = "http/protobuf";
 
   static String getOtlpProtocol(String dataType, ConfigProperties config) {
-    String protocol = config.getString("otel.exporter.otlp." + dataType + ".protocol");
-    if (protocol == null) {
-      protocol = config.getString("otel.exporter.otlp.protocol");
-    }
-    return (protocol == null) ? PROTOCOL_GRPC : protocol;
+    String configKey = "otel.exporter.otlp." + dataType + ".protocol";
+    String fallback = config.getString("otel.exporter.otlp.protocol", PROTOCOL_GRPC);
```

Comment:
This seems harder to read then previously as usually you don't pre-initialize a fallback and the variable names are not semantic anymore. I'd probably use the nullable version for the first lookup, for all the similar fallback type of lookups here.

```java
String protocol = config.getString("otel.exporter.otlp." + dataType + ".protocol);
if (protocol != null) {
  return protocol;
}
return config.getString("otel.exporter.otlp.protocol", PROTOCOL_GRPC);
```

Reply:
Yeah, the starting point had both the conditional and the ternary. This seems like a good compromise. 👍🏻